packages/cli/src/commands/new-list.test.ts

```diff
@@ -0,0 +1,320 @@
+import assert from 'node:assert';
+import { getMockedConfig } from '../tests/config.js';
+import {
+  assertStorageLocked,
+  assertStorageLogged,
+  assertStorageUnlocked,
+  assertStorageWaited,
+} from '../tests/storage.js';
+import { assertCommandDone, assertCommandFailed } from '../tests/plugin.js';
+import { CommandAbortError } from '../errors.js';
+import { listCommand } from './new-list.js';
+
+describe('new list command', () => {
+  const doneMigration = mock.fn(async function doneMigration() {});
+  const pendingMigration1 = mock.fn(async function pendingMigration1() {});
+  const pendingMigration2 = mock.fn(async function pendingMigration2() {});
+  const pendingMigration3 = mock.fn(async function pendingMigration3() {});
+  const migrationError = new Error('Migration failed');
+  const failedMigration = mock.fn(async function failedMigration() {});
+
+  beforeEach(() => {
+    doneMigration.mock.resetCalls();
+    pendingMigration1.mock.resetCalls();
+    pendingMigration2.mock.resetCalls();
+    pendingMigration3.mock.resetCalls();
+    failedMigration.mock.resetCalls();
+  });
+
+  /**
+   * LISTING MIGRATIONS
+   */
+  describe('listing migrations', () => {
+    it('does not execute any migrations', async () => {
+      // Given
+      const { config } = getMockedConfig([
+        [doneMigration, 'done'],
+        [pendingMigration1, 'pending'],
+        [pendingMigration2, 'pending'],
+      ]);
+
+      // When
+      await listCommand(config);
+
+      // Then
+      assert.strictEqual(doneMigration.mock.callCount(), 0);
+      assert.strictEqual(pendingMigration1.mock.callCount(), 0);
+      assert.strictEqual(pendingMigration2.mock.callCount(), 0);
+    });
+
+    it('lists all migrations regardless of state', async () => {
+      // Given
+      const { config, plugin } = getMockedConfig([
+        [doneMigration, 'done'],
+        [pendingMigration1, 'pending'],
+        [failedMigration, 'failed', migrationError],
+        [pendingMigration2, 'pending'],
+      ]);
+
+      // When
+      await listCommand(config);
+
+      // Then
+      assertCommandDone(plugin, [
+        [doneMigration.name, 'done'],
+        [pendingMigration1.name, 'pending'],
+        [failedMigration.name, 'failed', migrationError],
+        [pendingMigration2.name, 'pending'],
+      ]);
+    });
+
+    it('lists migrations in the registered order', async () => {
+      // Given
+      const finishOrder: string[] = [];
+      const { config, plugin } = getMockedConfig([
+        [pendingMigration1, 'pending'],
+        [pendingMigration2, 'pending'],
+        [pendingMigration3, 'pending'],
+      ]);
+
+      // Capture the order by inspecting the plugin callback
+      const originalDoneHook = plugin.hooks['emigrate:migration:done'];
+      plugin.hooks['emigrate:migration:done'] = mock.fn((parameters) => {
+        finishOrder.push(parameters.migration.identifier);
+        return originalDoneHook(parameters);
+      });
```

Comment:
As `plugin.hooks['emigrate:migration:done']` already is a mock function you can use its `.mockImplementation` method instead of having to use the `originalDoneHook` variable